core/include/detray/core/detector.hpp

```diff
@@ -444,6 +454,11 @@ class detector {
         return _surfaces_finder;
     }
 
+    DETRAY_HOST_DEVICE
+    inline const auto &get_n_max_objects_per_volume() {
```

Comment:
Right. I put const qualifier

Reply:
To be frank, we should not be returning references to member variables period, if we can at all avoid it. :eyes:

Reply:
:thinking: Yeah... when I first made the comment, I thought that the return value could be not just a simple integer, but that it could be an array (or something even more complicated) as well.

If it can only ever be an integer, then that should of course be returned by value. Better yet, for that we should use an explicit type declaration in the function's signature.

At one point I have the plan of going through the code and to eliminate all `auto`-s that do not have a good enough reason to be there. But since this is a new function, we might as well not make it use `auto` in the first place...

Reply:
I just removed reference and `auto` :p